marconi-mamba/db-utils/src/Marconi/DB/SqlUtils.hs

```diff
@@ -42,22 +45,27 @@ instance (FromField a) => FromRow (ShelleyFrequencyTable a) where
 instance (ToField a) => ToRow (ShelleyFrequencyTable a )where
   toRow (ShelleyFrequencyTable ad f) = toRow(ad, f)
 
+instance FromField C.AddressAny where
+  fromField f = fromField f >>=
+      maybe (returnError ConversionFailed f "Cannot deserialise address.")
+          pure . C.deserialiseFromRawBytes C.AsAddressAny
+
 -- | create a small SQL pipeline:
 -- first create a table of addresses and their coresponding utxo counts.
 -- Next, create the shelleyaddresses table
 --
-freqUtxoTable :: DBQueryEnv -> IO ()
+freqUtxoTable :: DBEnv -> IO ()
 freqUtxoTable env =
-    void $ withQueryAction (env ^. dbConf) ( \conn ->
+    void $ withQueryAction env ( \conn ->
         execute_ conn "drop table if exists frequtxos"
         >> execute_ conn "drop table if exists shelleyaddresses"
         >> execute_ conn "create table frequtxos as select address, count (address) as frequency from utxos group by address order by frequency DESC"
         >> execute_ conn "delete from frequtxos where frequency < 50" -- we only want `intersing` data
         >> execute_ conn
            "create TABLE shelleyaddresses (address text not null, frequency int not null)")
 
-withQueryAction :: DBConfig -> (Connection -> IO a) -> IO a
-withQueryAction conf action =
+withQueryAction :: DBEnv -> (Connection -> IO a) -> IO a
```

Comment:
I'll need to take a close  look at this and will update

Reply:
The main purpose of this funciton is create a SQL pipeline and eventually create list of addresses with most number of utxos. Depending on the size of the utxo table. The table size can drastically change based on the run time cardano network. Consequently this function can take a few seconds or several minutes to complete. The pupose of the `withQueryAction`  is to show the elapsed time for this  job. I thought might be valuable to optimize the job so that it would not run out of memory. Since this is utility, I have not spent much time mamanging the SQLite memory consumption.  So I used the elapsed time as an indicator. 
the correct solution is to use the SQLite query result streaming,[ fold and fold_](https://hackage.haskell.org/package/sqlite-simple-0.4.18.2/docs/Database-SQLite-Simple.html#g:14),  to reduce the memory foot print of SQL